test/test.groovy

```diff
@@ -0,0 +1,159 @@
+// Method definitions
+//************************************************************
+
+/*
+ * Parse command line arguments and store in Build Properties
+ */
+def loadBuildProperties(String [] args) {
+	// use CliBuilder to parse test.groovy input options
+	def cli = new CliBuilder(
+	   usage: '$DBB_HOME/bin/groovyz test.groovy <options>',
+	   header: '\nAvailable options (use -h for help):\n',
+	   footer: '\nInformation provided via above options is used to execute build against ZAppBuild.\n')
+	
+	cli.with
+	{
+	   h(longOpt: 'help', 'Show usage information')
+	   
+	   // test framework options
+	   b(longOpt: 'branch', 'zAppBuild branch to test', args: 1, required: true)
+
+	   // zAppBuild options
+	   a(longOpt: 'app', 'Application that is being tested (example: MortgageApplication)', args: 1, required: true)
+	   q(longOpt: 'hlq', 'HLQ for dataset reation / deletion (example: USER.BUILD)', args: 1, required: true)
+	   u(longOpt: 'url', 'DBB Web Application server URL', args: 1, required: true)
+	   i(longOpt: 'id', 'DBB Web Application user id', args: 1, required: true)
+	   p(longOpt: 'pw', 'DBB Web Application user password', args: 1)
+	   P(longOpt: 'pwFile', 'DBB Web Application user password file', args: 1)
+	   v(longOpt: 'verbose', 'Flag indicating to print trace statements')
+	   f(longOpt: 'propFiles', 'Commas spearated list of additional property files to load. Absolute paths or relative to workspace', args:1)
+           o(longOpt: 'outDir', 'Absolute path to the build output root directory', args:1)
+	}
+	
+	def options = cli.parse(args)
+	
+	// show usage text when -h or --help option is used.
+	if (options.h) {
+		cli.usage()
+		System.exit(0)
+	}
+	
+	// load build properties
+	BuildProperties props = BuildProperties.getInstance()
+	
+	// store the command line arguments in the Build Properties for all scripts to access
+	if (options.b) props.branch = options.b
+	if (options.a) props.app = options.a
+	if (options.q) props.hlq = options.q
+	if (options.u) props.url = options.u
+	if (options.i) props.id = options.i
+	if (options.p) props.pw = options.p
+	if (options.P) props.pwFile = options.P
+	if (options.v) props.verbose = 'true'
+	if (options.f) props.propFiles = options.f
+	if (options.o) props.outDir = options.o
+	
+	// load application test.properties file
+	props.load(new File("${getScriptDir()}/applications/${props.app}/test.properties"))
+	
+	// add some additional properties
+	props.testBranch = 'zAppBuildTesting'
+	props.zAppBuildDir = new File(getScriptDir()).getParent()
+	
+	// load property files from argument list
+	if (options.f) props.propFiles = options.f
+	if (props.propFiles) {
+		String[] propFiles = props.propFiles.split(',')
+		propFiles.each { propFile ->
+			if (!propFile.startsWith('/'))
+				propFile = "${props.workspace}/${propFile}"
+
+			if (options.v) println "** Loading property file ${propFile}"
+			props.load(new File(propFile))
+		}
+	}
+	
+	// zAppBuild repo locations
```

Comment:
Maybe a thought for the future enhancement of the test framework. Right now, we assume, that the application is part of zAppBuild, so we limit it to MortgageApplication for now. In the future we might would allow also additional applications to run the test against.